src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -85,7 +85,7 @@
    :values facts/fact-set
    :timestamp pls/Timestamp
    :environment (s/maybe s/Str)
-   :producer_timestamp (s/either (s/maybe s/Str) pls/Timestamp)})
+   :producer_timestamp (s/either s/Str pls/Timestamp)})
```

Comment:
prexisting obviously, but I'm not sure we actually have a reason to accept strings here